src/main/java/edu/ksu/canvas/attendance/form/CourseConfigurationValidator.java

```diff
@@ -14,28 +14,40 @@ public boolean supports(Class<?> clazz) {
 
     @Override
     public void validate(Object target, Errors errors) {
+
         CourseConfigurationForm courseConfigurationForm = (CourseConfigurationForm) target;
+
+        String assignmentPoints = courseConfigurationForm.getAssignmentPoints();
+        String presentPoints = courseConfigurationForm.getPresentPoints();
+        String tardyPoints = courseConfigurationForm.getTardyPoints();
+        String absentPoints = courseConfigurationForm.getAbsentPoints();
+        String excusedPoints = courseConfigurationForm.getExcusedPoints();
+
+
         if (courseConfigurationForm.getTotalClassMinutes() < courseConfigurationForm.getDefaultMinutesPerSession() && !errors.hasFieldErrors("totalClassMinutes")) {
             errors.rejectValue("defaultMinutesPerSession", "ExceedTotal.courseConfigurationForm.defaultMinutesPerSession");
         }
 
         if(courseConfigurationForm.getGradingOn()) {
             if (courseConfigurationForm.getAssignmentName() == null || courseConfigurationForm.getAssignmentName().length() <= 1 || courseConfigurationForm.getAssignmentName().trim().isEmpty()) {
                 errors.rejectValue("assignmentName", "Assignment Name is required.");
+                return;
             }
 
-            if (courseConfigurationForm.getAssignmentPoints() == null || courseConfigurationForm.getAssignmentPoints() <= 0) {
+            if (assignmentPoints == null || (Double.parseDouble(assignmentPoints) < 0)) {
                 errors.rejectValue("assignmentPoints", "Total Points is a required field and must be greater than 0.");
+                return;
             }
 
-            if (courseConfigurationForm.getPresentPoints() == null || courseConfigurationForm.getTardyPoints() == null || courseConfigurationForm.getExcusedPoints() == null
-                    || courseConfigurationForm.getAbsentPoints() == null) {
+            if (presentPoints == null || tardyPoints == null || absentPoints == null || excusedPoints == null) {
                 errors.rejectValue("presentPoints", "All status point fields are required.");
                 return;
             }
 
-            if ((courseConfigurationForm.getPresentPoints() > 100 || courseConfigurationForm.getPresentPoints() < 0) || (courseConfigurationForm.getTardyPoints() > 100 || courseConfigurationForm.getTardyPoints() < 0)
-                    || (courseConfigurationForm.getExcusedPoints() > 100 || courseConfigurationForm.getExcusedPoints() < 0) || (courseConfigurationForm.getAbsentPoints() > 100 || courseConfigurationForm.getAbsentPoints() < 0)) {
+            if ((Double.parseDouble(presentPoints) > 100) || (Double.parseDouble(presentPoints) < 0)
```

Comment:
maybe create a small helper method for converting to double and checking a string number is out of bounds that returns a boolean... like:
private boolean isValueOutOfBounds(String value) {
     Double val = Double.parseDouble(val);
     return val > 100 || val < 0;
}

that would cut down the number of conditions to 4 with 3 operators and might fix this sonar issue... like: 
if(isValueOutOfBounds(presentPoints) || isValueOutOfBounds(tardyPoints) || isValueOutOfBounds(absentPoints) || isValueOutOfBounds(excusedPoints)){// do stuff }